addons/common/functions/fnc_disableUserInput.sqf

```diff
@@ -16,76 +18,93 @@ private ["_state", "_dlg"];
 _state = _this select 0;
 
 if (_state) then {
-  disableSerialization;
+    disableSerialization;
 
-  if (!isNull (uiNamespace getVariable [QGVAR(dlgDisableMouse), displayNull])) exitWith {};
-  if ("ACE_DisableUserInput" in ([BIS_stackedEventHandlers_onEachFrame, {_this select 0}] call FUNC(map))) exitWith {};
+    if (!isNull (uiNamespace getVariable [QGVAR(dlgDisableMouse), displayNull])) exitWith {};
+    if ("ACE_DisableUserInput" in ([BIS_stackedEventHandlers_onEachFrame, {_this select 0}] call FUNC(map))) exitWith {};
 
-  // end TFAR and ACRE2 radio transmissions
-  0 spawn FUNC(endRadioTransmission);
+    // end TFAR and ACRE2 radio transmissions
+    call FUNC(endRadioTransmission);
 
-  // Close map
-  if (visibleMap && {!(player getVariable ["ACE_canSwitchUnits", false])}) then {openMap false};
-
-  closeDialog 0;
-  createDialog QGVAR(DisableMouse_Dialog);
-
-  _dlg = uiNamespace getVariable QGVAR(dlgDisableMouse);
+    // Close map
+    if (visibleMap && {!(player getVariable ["ACE_canSwitchUnits", false])}) then {
+        openMap false;
+    };
 
-  _dlg displayAddEventHandler ["KeyDown", {
-    _key = _this select 1;
+    closeDialog 0;
```

Comment:
You probably want this line to be:

```
while {dialog} do {
    closeDialog 0;
};
```

If a child dialog is open then this will keep closing the dialog until all parent dialogs are also closed.